File: backend/webdav/src/main/java/com/fsck/k9/backend/webdav/CommandMarkAllAsRead.java
package com.fsck.k9.backend.webdav;


import java.util.Collections;

import com.fsck.k9.mail.Flag;
import com.fsck.k9.mail.Folder;
import com.fsck.k9.mail.MessagingException;
import com.fsck.k9.mail.store.webdav.WebDavFolder;
import com.fsck.k9.mail.store.webdav.WebDavStore;
import org.jetbrains.annotations.NotNull;


class CommandMarkAllAsRead {
    private final WebDavStore webDavStore;


    CommandMarkAllAsRead(WebDavStore webDavStore) {
        this.webDavStore = webDavStore;
    }

    void markAllAsRead(@NotNull String folderServerId) throws MessagingException {
        WebDavFolder remoteFolder = webDavStore.getFolder(folderServerId);
        if (!remoteFolder.exists()) {
            return;
        }

        try {
            remoteFolder.open(Folder.OPEN_MODE_RW);
            if (remoteFolder.getMode() != Folder.OPEN_MODE_RW) {
                return;
            }

            remoteFolder.setFlags(Collections.singleton(Flag.SEEN), true);
        } finally {
            remoteFolder.close();
        }
    }
}

File: backend/imap/src/main/java/com/fsck/k9/backend/imap/CommandMarkAllAsRead.kt
package com.fsck.k9.backend.imap

import com.fsck.k9.mail.store.imap.ImapStore
import com.fsck.k9.mail.store.imap.OpenMode
import net.thunderbird.core.common.mail.Flag

internal class CommandMarkAllAsRead(private val imapStore: ImapStore) {

    fun markAllAsRead(folderServerId: String) {
        val remoteFolder = imapStore.getFolder(folderServerId)
        try {
            remoteFolder.open(OpenMode.READ_WRITE)

            remoteFolder.setFlagsForAllMessages(setOf(Flag.SEEN), true)
        } finally {
            remoteFolder.close()
        }
    }
}
